syntax = "proto3";

package protobuf;
option go_package = "github.com/bic4907/webrtc/protobuf";

service Service {
  rpc SendMessage (ReqMessage) returns (RplMessage);
  rpc StreamVideo (stream VideoChunk) returns (ReceiveReply);
}

message ReqMessage {
  string Content = 1;
}
message RplMessage {
  string Content = 1;
}

message VideoChunk {
  string  RoomId      = 1;
  string  UserId      = 2;
  bytes   Chunk       = 3;
  int64   CreatedAt   = 4;
}

message ReceiveReply {
  string Result = 1;
}